src/main/java/com/xatkit/core/recognition/processor/ToxicityPostProcessor.java

```diff
@@ -0,0 +1,93 @@
+package com.xatkit.core.recognition.processor;
+
+import com.xatkit.execution.StateContext;
+import com.xatkit.intent.RecognizedIntent;
+import org.apache.commons.configuration2.Configuration;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Analyzes the probability of a comment being toxic with different types of toxicity attributes
+ */
+public class ToxicityPostProcessor implements IntentPostProcessor {
+
+    /**
+     * The PerspectiveAPI Client that will ask the requests to PerspectiveAPI
+     */
+    private PerspectiveapiInterface perspectiveapiClient;
+    private DetoxifyInterface detoxifyClient;
+
+    /**
+     * The context parameter key used to store the toxicity attributes extracted from the user input.
+     */
+    protected final static String PERSPECTIVEAPI_PARAMETER_KEY = "nlp.perspectiveapi.";
+    protected final static String DETOXIFY_PARAMETER_KEY = "nlp.detoxify.original.";
+
+    /**
+     * Instantiates a new Toxicity post processor.
+     *
+     * @param configuration the Xatkit bot configuration that contains the ToxicityPostProcessor parameters
+     */
+    public ToxicityPostProcessor(Configuration configuration) {
+        this.perspectiveapiClient = null;
+        this.detoxifyClient = null;
+        boolean xatkit_perspectiveapi = configuration.getProperty("xatkit.perspectiveapi") != null &&
+                (boolean) configuration.getProperty("xatkit.perspectiveapi");
+        boolean xatkit_detoxify = configuration.getProperty("xatkit.detoxify") != null &&
+                (boolean) configuration.getProperty("xatkit.detoxify");
+        if (xatkit_perspectiveapi) {
+            String apiKey = (String) configuration.getProperty("xatkit.perspectiveapi.apiKey");
+            String language = (String) configuration.getProperty("xatkit.perspectiveapi.language");
+            Boolean doNotStore = (Boolean) configuration.getProperty("xatkit.perspectiveapi.doNotStore");
+            String clientToken = (String) configuration.getProperty("xatkit.perspectiveapi.clientToken");
+            String sessionId = (String) configuration.getProperty("xatkit.perspectiveapi.sessionId");
+            this.perspectiveapiClient = new PerspectiveapiInterface(apiKey, language, doNotStore, clientToken,
+                    sessionId,
+                    PERSPECTIVEAPI_PARAMETER_KEY);
```

Comment:
As stated above: no need to pass this information to the interface. We can take care of converting raw results into context parameter in this class.